Conceptpower+Spring/src/main/java/edu/asu/conceptpower/web/ConceptSearchController.java

```diff
@@ -24,43 +31,47 @@
 @Controller
 public class ConceptSearchController {
 
-	@Autowired
-	private IConceptManager conceptManager;
+    @Autowired
+    private IConceptManager conceptManager;
+
+    @Autowired
+    private IConceptWrapperCreator wrapperCreator;
+
+    @Autowired
+    private ConceptSearchValidator validator;
+
+    @InitBinder
+    private void initBinder(WebDataBinder binder) {
+        binder.setValidator(validator);
+    }
+
+    /**
+     * This method is searches concepts a specific term and pos
+     * 
+     * @param req
+     *            Holds the HTTP request information
+     * @param model
+     *            Generic model holder for servlet
+     * @return Returns a string value to redirect user to concept search page
+     */
+    @RequestMapping(value = "/home/conceptsearch", method = RequestMethod.GET)
+    public String search(HttpServletRequest req, ModelMap model,
+            @Validated @ModelAttribute("conceptSearchBean") ConceptSearchBean conceptSearchBean,
+            BindingResult results) {
+
+        if (results.hasErrors()) {
+            return "conceptsearch";
+        }
+
+        ConceptEntry[] found = conceptManager.getConceptListEntriesForWord(conceptSearchBean.getWord(),
+                conceptSearchBean.getPos().toLowerCase().trim());
+        List<ConceptEntryWrapper> foundConcepts = wrapperCreator.createWrappers(found);
+        conceptSearchBean.setFoundConcepts(foundConcepts);
+        if (CollectionUtils.isEmpty(foundConcepts)) {
+            results.rejectValue("foundConcepts", "no.searchResults");
+        }

```

Comment:
I wouldn't do it that way. It's not a validation error that there are no results. Just show a message "Nothing matching your search query." where the table usually would be whenever the results are 0.